docs/src/test/java/docs/http/javadsl/server/directives/MarshallingDirectivesExamplesTest.java

```diff
@@ -58,4 +63,58 @@ public void testEntity() {
     ).assertEntity("Person:Jane - favoriteNumber:42");
     //#example-entity-with-json
   }
+
+  @Test
+  public void testCompleteWith() {
+    //#example-completeWith-with-json
+    final Marshaller<Person, HttpResponse> marshaller = Marshaller.entityToOKResponse(Jackson.<Person>marshaller());
+
+    final Consumer<Consumer<Person>> findPerson = completionFunction -> {
+
+      //... some processing logic...
```

Comment:
You could even pass the `completionFunction` to a different thread here. Maybe mention that in a comment as well?